test/unit/lodash.spec.js

```diff
@@ -0,0 +1,605 @@
+                expect(element2).to.be.a('function');
+                expect(params).to.have.length(3);
+                var acc = params[2];
+                expect(acc).to.eql([]);
+                expect(_.isEmpty(acc)).to.eql(true);
+            });
+            it('should contain array with 5 Integer values', function() {
+                var array = params[0];
+                expect(array.length).to.eql(5);
+                array.forEach(function(key) {
+                    expect(Number.isInteger(key)).to.eql(true);
+                });
+            });
+            it('should contain function with 2 parameters', function() {
+                var fun = params[1];
+                expect(fun).to.have.length(2);
+            });
+
+            it('should return accumulated value', function() {
+                expect(_.transform.apply(_, params)).to.eql(['3', '5', '7']);
+            });
+        });
+
+        describe('transform2', function() {
+            beforeEach(function() {
+                params = datasets.transform2();
+            });
+            it('should match types of passing elements', function() {
+                var element1 = params[0] instanceof Object;
+                var element2 = params[1] instanceof Function;
+                var acc = params[2];
+                expect(element1).to.eql(true);
+                expect(element2).to.eql(true);
+                expect(params).to.have.length(3);
+                expect(acc).to.eql({});
+                expect(_.isEmpty(acc)).to.eql(true);
+            });
+            it('should contain 2 own properties', function() {
+                var obj = params[0];
+                expect(Object.keys(obj).length).to.eql(3);
+            });
+            it('should contain function with 3 parameters', function() {
+                var fun = params[1];
+                expect(fun).to.have.length(3);
+            });
+
+            it('should return accumulated value', function() {
+                expect(_.transform.apply(_, params)).to.eql({
+                    '5': ['width'],
+                    '10': ['weight'],
+                    '600': ['height']
+                });
+            });
+        });
+    });
+
+    describe('unset', function() {
+        describe('unset1', function() {
+            beforeEach(function() {
+                params = datasets.unset1();
+            });
+            it('should match types of passing elements', function() {
+                var element1 = params[0] instanceof Object;
+                var element2 = typeof params[1];
+                expect(element1).to.eql(true);
+                expect(element2).to.eql('string');
+                expect(params.length).to.eql(2);
+            });
+            it('should remove 4 from height', function() {
+                var obj = params[0];
+                var pattern = params[1];
+                expect(obj.rectangle[0].square.height).to.eql(4);
+                _.unset(obj, pattern);
```

Comment:
How it is possible that you haven't actually checked if property is deleted even that API clearly suggests returning value for that function??